src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -1042,6 +960,81 @@ private void TraverseScopeBranch(ParseNode node)
+        {
+            if (boilerplateLoadAndRunEntryLabel != "not-set-yet")
+                return boilerplateLoadAndRunEntryLabel;
+            
+            // Build the boilerplate function and return the entry point to it:
+            // ----------------------------------------------------------------
+            
+            List<Opcode> prevCurrentCodeSection = currentCodeSection; // for restoring it later down at the bottom.
+
+            // This boilerplate needs to be inserted into the functions section of the program:
+            Subprogram subprogramObject = context.Subprograms.GetSubprogram("boilerplate-load-and-run-func");
+            currentCodeSection = subprogramObject.FunctionCode;            
+
+            boilerplateLoadAndRunEntryLabel = GetNextLabel(false);
+            AddOpcode(new OpcodePushScope(-999,0)); // hardcoded dummy scope ID, the parent scope is the global.
+
+            // High level kerboscript function calls flip the argument orders for us, but
+            // low level kRISC does not so the parameters have to be read in stack order:
+            
+            // store parameter 2 in a local name:
+            AddOpcode(new OpcodePush("$runonce"));
+            AddOpcode(new OpcodeSwap());
+            AddOpcode(new OpcodeStoreLocal());
+
+            // store parameter 1 in a local name:
+            AddOpcode(new OpcodePush("$filename"));
+            AddOpcode(new OpcodeSwap());
+            AddOpcode(new OpcodeStoreLocal());
+            
+            // Unconditionally call load() no matter what.  load() will abort and return
+            // early if the program was already compiled, and tell us that on the stack:
+            AddOpcode(new OpcodePush(new KOSArgMarkerType()));
+            AddOpcode(new OpcodePush("$filename"));
+            AddOpcode(new OpcodeEval());
+            AddOpcode(new OpcodePush(true)); // the flag that tells load() to abort early if it's already loaded:
+            AddOpcode(new OpcodePush(null));
+            AddOpcode(new OpcodeCall("load()"));
+
+            // Stack now has the 2 return values of load():
+            //    Topmost value is a boolean flag for whether or not the program was already loaded.
+            //    Second-from-top value is the entry point into the program.
+
+            // If load() didn't claim the program was already loaded, or if we aren't operating
+            // in "once" mode, then jump to the part where we call the loaded program, else
+            // fall through to a dummy do-nothing return for the "run once, but it already ran" case:
+            Opcode branchFromOne = new OpcodeBranchIfFalse();
+            AddOpcode(branchFromOne);
+            AddOpcode(new OpcodePush("$runonce"));
+            Opcode branchFromTwo = new OpcodeBranchIfFalse();
+            AddOpcode(branchFromTwo);
+            AddOpcode(new OpcodePush(0));   // ---+-- The dummy do-nothing return.
```

Comment:
At this point, the "already loaded" flag has been popped, but the "entry point" is still on the stack.  So if "already loaded" is true and "$runonce" is true, we push "0" and return, won't that leave "entry point" on the stack?